In [26]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

<b></a><div class="alert alert-block alert-info">Задание 1

In [14]:
#задаем исходную матрицу потребления и производства
ce = np.array([[150, 180, 80],
            [75, 270, 40],
            [25, 60, 40]])

#создаем единичную матрицу
ed_matrix = np.eye(3)

#задаем вектор конечного потребления
Y = np.array([30, 25, 35])

In [15]:
ce

array([[150, 180,  80],
       [ 75, 270,  40],
       [ 25,  60,  40]])

In [16]:
sum_rows = []
#суммируем элементы по каждой строке
row_sums = np.sum(ce, axis=1) + Y

#сумма по столбцам
columns_sums = np.sum(ce, axis=0)

In [17]:
#создаем матрицу прямых производственных затрат
A = ce / columns_sums
A

array([[0.6       , 0.35294118, 0.5       ],
       [0.3       , 0.52941176, 0.25      ],
       [0.1       , 0.11764706, 0.25      ]])

In [22]:
#умножаем вектор на вектор
coef = np.array([1.1, 1.5, 1.2])
new_prod = coef * row_sums
new_prod

array([484., 615., 192.])

In [23]:
E_A = ed_matrix - A
E_A

array([[ 0.4       , -0.35294118, -0.5       ],
       [-0.3       ,  0.47058824, -0.25      ],
       [-0.1       , -0.11764706,  0.75      ]])

In [25]:
new_Y = E_A.dot(new_prod.reshape(3,1))
new_Y

array([[-119.45882353],
       [  96.21176471],
       [  23.24705882]])

In [ ]:
#Объединяем в единый код

ce = np.array([[150, 180, 80],
            [75, 270, 40],
            [25, 60, 40]]) #задаем исходную матрицу потребления и производства

ed_matrix = np.eye(3) #создаем единичную матрицу
Y = np.array([30, 25, 35]) #задаем вектор конечного потребления

sum_rows = []
row_sums = np.sum(ce, axis=1) + Y #суммируем элементы по каждой строке
columns_sums = np.sum(ce, axis=0) #сумма по столбцам

A = ce / columns_sums #создаем матрицу производственных затрат 

#умножаем вектор на вектор
coef = np.array([1.1, 1.5, 1.2])
new_prod = coef * row_sums

E_A = ed_matrix - A #находим разность единичной матрицы и м-цы производственных затрат
new_Y = E_A.dot(new_prod.reshape(3,1)) #находим новое конечное потребление

</a><div class="alert alert-block alert-warning"> Можем заметить, что полученное значение конечного потребления для 1-ой отрасли отрицательно (-119,5). Это может говорить о том, что увеличение валового выпуска на заданные проценты оказалось слишком большим, что привело к отрицательному конечному потреблению. Также стоит отметить, что полученная матрица Е_А является вырожденной (детерминанат мат-цы равен 0), что подтверждает идею того, что производственные затраты слишком велики по сравнению с производственными возможностями.

<b></a><div class="alert alert-block alert-info">Задание 2

In [27]:
#задаем матрицу прямых производственных затрат
A = np.array([[0.2, 0.1, 0.3],
            [0.1, 0.3, 0.2],
            [0.3, 0.2, 0.1]])

ed_matrix = np.eye(3) #создаем единичную матрицу

Y = np.array([155, 105, 40]) #задаем вектор конечного потребления

E_A = ed_matrix - A #находим разность единичной матрицы и м-цы производственных затрат

matrix_obr = np.linalg.inv(E_A) #создаем обратную матрицу

val_vypusk = matrix_obr.dot(Y.reshape(3,1)) #перемножаем матрицу на вектор
val_vypusk

array([[300.],
       [250.],
       [200.]])

</a><div class="alert alert-block alert-warning"> Полученные значения валового выпуска составили: для первой отрасли - 300, для второй - 250, для третьей - 200